datastore/datastore.go

```diff
@@ -40,27 +40,22 @@ type Datastore struct {
 	memcached *Memcached
 	db        database.IDatabaseService
 
-	knownValidatorsByPubkey   map[common.PubkeyHex]uint64
-	knownValidatorsByIndex    map[uint64]common.PubkeyHex
-	knownValidatorsLock       sync.RWMutex
+	knownValidatorsByPubkey sync.Map // map[common.PubkeyHex]uint64
+	knownValidatorsByIndex  sync.Map // map[uint64]common.PubkeyHex
+	validatorRegistrations  sync.Map // map[common.PubkeyHex]builderApiV1.ValidatorRegistration
```

Comment:
Do you think it might be better to implement small generic wrapper over sync.Map to have it typed? We use only like Load+Store, so it's going to be like 15 lines:

```go
type syncMap[K comparable, V any] struct {
	m sync.Map
}

func (m *syncMap[K, V]) Load(key K) (value V, ok bool) {
	v, ok := m.m.Load(key)
	if !ok {
		return value, ok
	}
	return v.(V), ok
}

func (m *syncMap[K, V]) Store(key K, value V) {
	m.m.Store(key, value)
}
```